File: src/test/java/im/tox/gui/FileTransfer.java
package im.tox.gui;

import java.io.Closeable;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.IOException;

public abstract class FileTransfer implements Closeable {

  protected final File file;

  public FileTransfer(File file) {
    this.file = file;
  }

  public abstract void resume() throws FileNotFoundException;

  public abstract byte[] read(long position, int length) throws IOException;

  public abstract void close() throws IOException;

  public abstract void write(long position, byte[] data) throws IOException;

}

File: projects/tox4j/src/test/java/im/tox/gui/domain/FileTransfer.scala
package im.tox.gui.domain

import java.io.{Closeable, File, FileNotFoundException, IOException}

abstract class FileTransfer(val file: File) extends Closeable {
  @throws[FileNotFoundException]
  def resume()

  @throws[IOException]
  def read(position: Long, length: Int): Array[Byte]

  @throws[IOException]
  def close()

  @throws[IOException]
  def write(position: Long, data: Array[Byte])
}
